### PR TITLE
Simplify CMapTable implementation

### PR DESCRIPTION
**Pull request by 0xced**
After reading the code of the `CMapTable` class for diagnosing #191 it appears that the code can be improved to more easily understandable.

### CODE REVIEW

File: src/SixLabors.Fonts/Tables/General/CMapTable.cs [hunk 1]
```diff
@@ -13,46 +13,29 @@ namespace SixLabors.Fonts.Tables.General
...
-                // keep looking until we have an index thats not the fallback.
-                if (t.TryGetGlyphId(codePoint, out glyphId))
-                {
-                    return true;
-                }
-            }
-
-            // didn't have a windows match just use any and hope for the best
-            foreach (CMapSubTable t in this.Tables)
+            foreach (CMapSubTable t in this.Tables.OrderBy(t => GetPreferredPlatformOrder(t.Platform)))
```
Comment (tocsoft) [inline src/SixLabors.Fonts/Tables/General/CMapTable.cs:36]: we will want to do the sort once in the constructor rather than on every glyph lookup.

Comment (0xced) [reply src/SixLabors.Fonts/Tables/General/CMapTable.cs]: Good catch! Addressed in ab083e77a695947c7ec20719ad6c46abb6c3837f.

File: src/SixLabors.Fonts/Tables/General/CMapTable.cs [hunk 2]
```diff
@@ -13,46 +13,29 @@ namespace SixLabors.Fonts.Tables.General
     internal sealed class CMapTable : Table
     {
         internal const string TableName = "cmap";
-        private readonly CMapSubTable[] platformTables;
 
-        public CMapTable(CMapSubTable[] tables)
+        private static readonly Dictionary<PlatformIDs, int> PreferredPlatformOrder = new Dictionary<PlatformIDs, int>
```
Comment (JimBobSquarePants) [inline src/SixLabors.Fonts/Tables/General/CMapTable.cs:17]: On my phone but isn’t this already an enum? A simple cast will be much faster than Dictionary + TryGet

Comment (0xced) [reply src/SixLabors.Fonts/Tables/General/CMapTable.cs:17]: Indeed it's an enum but the `PreferredPlatformOrder` dictionary defines the preferred *order* of the platform ids, i.e. Windows first to preserve the semantics of the previous code.

Comment (JimBobSquarePants) [reply src/SixLabors.Fonts/Tables/General/CMapTable.cs:17]: Ah yep, I see now, nice!

**Review by JimBobSquarePants — APPROVED** [review]
This is much more readable now. Thanks!

**Pull request merged**